superset-frontend/src/components/Checkbox/Checkbox.test.tsx

```diff
@@ -16,46 +16,122 @@
+  disabled: false,
+  title: 'Checkbox title',
+  indeterminate: false,
+  children: 'Checkbox Label',
+};
 
-jest.mock('src/components/Checkbox/CheckboxIcons', () => ({
-  CheckboxChecked: () => <div data-test="mock-CheckboxChecked" />,
-  CheckboxUnchecked: () => <div data-test="mock-CheckboxUnchecked" />,
-}));
+describe('Checkbox Component', () => {
+  const asyncRender = (props = mockedProps) =>
+    waitFor(() => render(<Checkbox {...props} />));
 
-describe('when unchecked', () => {
-  test('renders the unchecked component', () => {
-    const { getByTestId } = render(
-      <Checkbox style={{}} checked={false} onChange={() => true} />,
-    );
-    expect(getByTestId('mock-CheckboxUnchecked')).toBeInTheDocument();
+  beforeEach(() => {
+    jest.clearAllMocks();
   });
-});
 
-describe('when checked', () => {
-  test('renders the checked component', () => {
-    const { getByTestId } = render(
-      <Checkbox style={{}} checked onChange={() => true} />,
-    );
-    expect(getByTestId('mock-CheckboxChecked')).toBeInTheDocument();
+  describe('Rendering', () => {
+    test('should render correctly', async () => {
+      const { container } = await asyncRender();
+      expect(container).toBeInTheDocument();
+    });
+
+    test('should render the label', async () => {
+      await asyncRender();
+      expect(screen.getByText('Checkbox Label')).toBeInTheDocument();
+    });
+
+    test('should render the checkbox', async () => {
+      await asyncRender();
+      expect(screen.getByRole('checkbox')).toBeInTheDocument();
+    });
   });
-});
 
-test('works with an onChange handler', () => {
-  const mockAction = jest.fn();
-  const { getByRole } = render(
-    <Checkbox style={{}} checked={false} onChange={mockAction} />,
-  );
-  fireEvent.click(getByRole('checkbox'));
-  expect(mockAction).toHaveBeenCalled();
-});
+  describe('States', () => {
+    test('should render as unchecked when checked is false', async () => {
+      await asyncRender();
+      const checkbox = screen.getByRole('checkbox');
+      expect(checkbox).not.toBeChecked();
+    });
+
+    test('should render as checked when checked is true', async () => {
+      const checkedProps = { ...mockedProps, checked: true };
+      await asyncRender(checkedProps);
+      const checkbox = screen.getByRole('checkbox');
+      expect(checkbox).toBeChecked();
+    });
+
+    test('should render as indeterminate when indeterminate is true', async () => {
+      const indeterminateProps = { ...mockedProps, indeterminate: true };
+      await asyncRender(indeterminateProps);
+      const checkbox = screen.getByRole('checkbox');
+      expect(checkbox).toBeInTheDocument();
+      expect((checkbox as HTMLInputElement).indeterminate).toBe(true);
+    });
+
+    test('should render as disabled when disabled prop is true', async () => {
+      const disabledProps = { ...mockedProps, disabled: true };
+      await asyncRender(disabledProps);
+      const checkboxWrapper = screen
```

Comment:
Can we use a `expect(screen.getByRole('checkbox')).toBeDisabled()`? With RTL we should rely on use behaviors not DOM structures as much possible.